application-task-default/src/main/java/com/xwiki/task/internal/notifications/taskchanged/TaskChangedEventNotificationListener.java

```diff
@@ -62,6 +63,8 @@ public class TaskChangedEventNotificationListener extends AbstractEventListener
 
     private static final EntityReference CLASS_MATCHER = BaseObjectReference.any("TaskManager.TaskManagerClass");
 
+    private static final BeginEvent FOLD_EVENT_MATCHER = event -> event instanceof BeginFoldEvent;
```

Comment:
Could you add a small comment mentioning why you listen to fold events?

Reply:
If a change is not related to the current issue, you shouldn't commit it, as it is misleading and affects the git history. I know some are minor changes, but these still need to be separated
If a change does not affect functionality, you can create a `[Misc]` commit. But even in this case, if there are a lot of changes, you should still create an issue. E.g. for functional tests, even if it doesn't affect functionality, you still need an issue, but if you update the `release.sh`, just a misc commit is enough

In this particular case, you have an actual issue from what you say, this being related to sending notifications on migrations